clippy_lints/src/methods/map_with_unused_argument_over_ranges.rs

```diff
@@ -0,0 +1,118 @@
+                }
+            } else {
+                0
+            };
+            return Some(format!("{count}"));
+        }
+        let end_snippet = snippet_with_applicability(cx, end.span, "...", applicability).into_owned();
+        if lower_bound == 0 {
+            if range.limits == RangeLimits::Closed {
+                return Some(format!("{end_snippet} + 1"));
+            }
+            return Some(end_snippet);
+        }
+        if range.limits == RangeLimits::Closed {
+            return Some(format!("{end_snippet} - {}", lower_bound - 1));
+        }
+        return Some(format!("{end_snippet} - {lower_bound}"));
+    }
+    None
+}
+
+pub(super) fn check(
+    cx: &LateContext<'_>,
+    ex: &Expr<'_>,
+    receiver: &Expr<'_>,
+    arg: &Expr<'_>,
+    msrv: &Msrv,
+    method_call_span: Span,
+) {
+    if !msrv.meets(msrvs::REPEAT_WITH) {
+        return;
+    }
+    let mut applicability = Applicability::MaybeIncorrect;
+    if let Some(range) = higher::Range::hir(receiver)
+        && let ExprKind::Closure(Closure { body, .. }) = arg.kind
+        && let body_hir = cx.tcx.hir().body(*body)
+        && let Body {
+            params: [param],
+            value: body_expr,
+        } = body_hir
+        && !usage::BindingUsageFinder::are_params_used(cx, body_hir)
+        && let Some(count) = extract_count_with_applicability(cx, range, &mut applicability)
+    {
+        if eager_or_lazy::switch_to_eager_eval(cx, body_expr) {
+            let body_snippet = snippet_with_applicability(cx, body_expr.span, "..", &mut applicability);
+            span_lint_and_then(
+                cx,
+                MAP_WITH_UNUSED_ARGUMENT_OVER_RANGES,
+                ex.span,
+                "map of a closure that does not depend on its parameter over a range",
+                |diag| {
+                    diag.multipart_suggestion(
+                        "remove the explicit range and use `repeat` and `take`",
+                        vec![
+                            (receiver.span.to(method_call_span), "std::iter::repeat".to_owned()),
+                            (arg.span, body_snippet.to_string()),
+                            (ex.span.shrink_to_hi(), format!(".take({count})")),
+                        ],
+                        applicability,
+                    );
+                },
+            );
+        } else {
+            span_lint_and_then(
+                cx,
+                MAP_WITH_UNUSED_ARGUMENT_OVER_RANGES,
+                ex.span,
+                "map of a closure that does not depend on its parameter over a range",
+                |diag| {
+                    diag.multipart_suggestion(
+                        "remove the explicit range and use `repeat_with` and `take`",
+                        vec![
+                            (receiver.span.to(method_call_span), "std::iter::repeat_with".to_owned()),
+                            (param.span, String::new()),
```

Comment:
The only difference between this branch and the `span_lint_and_then` above is the function to suggest (repeat/repeat_with) and which span to remove right?
Can we just select those two based on `switch_to_eager_eval()` and merge all the rest of the `two span_lint_and_then calls` so the lint messages don't need to be repeated?